quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/resources/DebugHostnameSettingsResource.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.quarkus.runtime.services.resources;
+
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.quarkus.runtime.Environment;
+import org.keycloak.quarkus.runtime.configuration.Configuration;
+import org.keycloak.services.Urls;
+import org.keycloak.services.resources.Cors;
+import org.keycloak.theme.FreeMarkerException;
+import org.keycloak.theme.Theme;
+import org.keycloak.theme.freemarker.FreeMarkerProvider;
+import org.keycloak.urls.UrlType;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+@Path("/realms")
+public class DebugHostnameSettingsResource {
```

Comment:
I don't think we should put the debug tool into the Quarkus module. It is generic enough, it's a debug tool for our hostname SPI. I'd put it among other REST endpoints [here](https://github.com/keycloak/keycloak/tree/main/services/src/main/java/org/keycloak/services/resources). WDYT?

Reply:
@vmuzikar ,

In my first attempt, I tried to put this class exactly where you are mentioning now. But due to the dependencies below, the solution I've found was to create the Resource under the Quarkus module.

```java
import org.keycloak.quarkus.runtime.Environment;
import org.keycloak.quarkus.runtime.configuration.Configuration;
````

Could you see another way to supply such needed dependencies from "keycloak-services" module, without touching the "pom.xml"?

Thanks.

Reply:
That's a good point. We certainly don't want to add Quarkus dependency to keycloak-services as it should be framework/app server agnostic. While the debug tool is focusing on the Hostname SPI (which is not specific to Quarkus), the config options that we are printing there are Quarkus specific.

So I believe this approach might be correct in the end.

@pedroigor WDYT?